### ISSUE TITLE
Custom result variable encoding types

### ISSUE THREAD

**Issue by J12934** [nikku/camunda-worker-node#14]
Hi!
First of all thank you for this package! 😉

I have unfortunately run into a use case in which I need more direct control on the type and serialization format of the result variables submitted by the worker function.

I'd like to add functionality to manually add type and serialization information to a result variable.

I'm currently thinking that it should look something like this:

```js
var Worker = require('camunda-worker-node');
var ManualSerialization = require('camunda-worker-node/lib/manual-serialization');

// ...

worker.subscribe('work:A', [], async function(context) {
    return {
        variables: {
            customVar: new ManualSerialization({
                value: '"foobar"',
                type: 'Object',
                valueInfo: {
                    objectTypeName: 'java.lang.String',
                    serializationDataFormat: 'application/json',
                },
            }),
        },
    };
});
```

The object passed into the ManualSerialization class would then get directly passed into the external task completion request without any magic.

Would this be a good addition or is there a better way to support this?

**Comment by nikku** [comment]
Looks like a valuable addition to the library. 

Would you mind adding a PR for it?

**Comment by J12934** [comment]
Ok cool!
Yup will send in a PR soon.

**Comment by meyerdan** [comment]
Plase also have a look at https://github.com/camunda/camunda-external-task-client-js

**Comment by nikku** [comment]
@meyerdan [Referenced in `README`](https://github.com/nikku/camunda-worker-node#related), by the way :wink:.

**Comment by J12934** [comment]
Oh that is cool that there is a official package now!
Is there any direct connection between the packages?

Will probably still stick to this package for now, should hopefully be simpler and more fun than to migrate 😉

**Comment by J12934** [comment]
I just saw this test: 
https://github.com/nikku/camunda-worker-node/blob/master/test/workers-spec.js#L874

Apparently custom variable serialization should already work fine without this addition.
Is this really working correctly? If so i will just add some docs about it instead of re implementing the feature in a different way.

**Comment by nikku** [comment]
Yes, custom variables work perfectly fine if it is just about keeping the inbound serialization format.

What's not possible right now is creating new variables with a custom format (i.e. to match certain Java classes). This is by design.

I welcome a contribution such as the one you've sketched as an optional extension though.

**Comment by nikku** [comment]
Thanks for your contribution. I've merged it and released `v7.1.0` with the new feature in place.

Featured [in the README](https://github.com/nikku/camunda-worker-node#variable-serialization), too :heart:.
